src/Pim/Component/Catalog/Factory/ProductValue/DateProductValueFactory.php

```diff
@@ -0,0 +1,61 @@
+<?php
+
+namespace Pim\Component\Catalog\Factory\ProductValue;
+
+use Pim\Component\Catalog\Model\AttributeInterface;
+
+/**
+ * Factory that creates date product values.
+ *
+ * @internal  Please, do not use this class directly. You must use \Pim\Component\Catalog\Factory\ProductValueFactory.
+ *
+ * @author    Damien Carcel (damien.carcel@akeneo.com)
+ * @copyright 2017 Akeneo SAS (http://www.akeneo.com)
+ * @license   http://opensource.org/licenses/osl-3.0.php Open Software License (OSL 3.0)
+ */
+class DateProductValueFactory implements ProductValueFactoryInterface
```

Comment:
Because we perform a `new \DateTime()` before `setDate`, so it has to be a dedicated factory.